mungegithub/mungers/close-stale-pr.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"time"
+
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const (
+	stalePullRequestHours = 24 * 30 * 3
+	// TODO: Explain how to prevent this from happening (and implement the solution)
+	closingComment = `This PR hasn't been active in a while.
+
+Please re-open if necessary`
+)
+
+type CloseStalePR struct{}
+
+func init() {
+	s := CloseStalePR{}
+	RegisterMungerOrDie(s)
+}
+
+// Name is the name usable in --pr-mungers
+func (CloseStalePR) Name() string { return "close-stale-pr" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (CloseStalePR) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (CloseStalePR) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (CloseStalePR) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (CloseStalePR) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+func lastHumanUpdatePullRequest(obj *github.MungeObject) (*time.Time, error) {
+	pr, err := obj.GetPR()
+	if err != nil {
+		return nil, err
+	}
+
+	comments, err := obj.ListPRComments()
+	if err != nil {
+		return nil, err
+	}
+
+	last_human := pr.CreatedAt
+	for i := range comments {
+		comment := comments[i]
+		if comment.User == nil || comment.User.Login == nil || comment.CreatedAt == nil || comment.Body == nil {
+			continue
+		}
+		if *comment.User.Login == botName || *comment.User.Login == jenkinsBotName {
+			continue
+		}
+		last_human = comment.UpdatedAt
```

Comment:
Good call, they're not. They're actually sorted by Id, which is usually going to lead to a chronological order ... unless you update an older comment.